e2e/tests/container_pause.go

```diff
@@ -0,0 +1,90 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package tests
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/runfinch/common-tests/command"
+	"github.com/runfinch/common-tests/option"
+
+	"github.com/runfinch/finch-daemon/api/response"
+	"github.com/runfinch/finch-daemon/e2e/client"
+)
+
+func ContainerPause(opt *option.Option) {
+	Describe("pause a container", func() {
+		var (
+			uClient *http.Client
+			version string
+			apiUrl  string
+		)
+
+		BeforeEach(func() {
+			uClient = client.NewClient(GetDockerHostUrl())
+			version = GetDockerApiVersion()
+			relativeUrl := fmt.Sprintf("/containers/%s/pause", testContainerName)
+			apiUrl = client.ConvertToFinchUrl(version, relativeUrl)
+		})
+
+		AfterEach(func() {
+			command.RemoveAll(opt)
+		})
+
+		It("should pause a running container", func() {
+			// Start a container that keeps running
+			command.Run(opt, "run", "-d", "--name", testContainerName, defaultImage, "sleep", "infinity")
+
+			res, err := uClient.Post(apiUrl, "application/json", nil)
+			Expect(err).Should(BeNil())
+			Expect(res.StatusCode).Should(Equal(http.StatusNoContent))
+
+			// Verify container is paused
+			output := command.StdoutStr(opt, "inspect", "--format", "{{.State.Status}}", testContainerName)
+			Expect(output).Should(ContainSubstring("paused"))
```

Comment:
Instead of a string match, you should verify the inspect response field of `Status` is set to "paused"